load spotify playlists from spotcast sensor

landonr opened the pull request:
for issue https://github.com/landonr/esphome-remote/issues/13
uses spotcast sensor https://github.com/fondberg/spotcast#use-the-sensor
use ArduinoJSON for decoding sensors with json
add TV to soundbar sources
fix soundbar not showing up in media players menu if it was playing tv 
move friendlyNameEntity to a new file and use it more

abroooo reviewed (commented):
Don't know if you wanted me to review but I just do it 😄  Looks good to me so far. I just have a few suggestions/questions though.
Maybe it's worth mentioning that I'm still not that familiar with a lot of parts in the code yet especially the MediaPlayer part.

I think it would also be good to update the Readme/Getting Started because it requires a certain configuration on the home assistant side.

landonr commented on esphomeRemote/FriendlyNameEntity.h:
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
i saw this in another project and it seemed cleaner because it's only one line

landonr commented:
@abroooo I appreciate the review! I moved all the text helpers to it's own class and made them static so we don't have to initialize the class. I also changed all the ifndef's to pragma once

landonr commented on esphomeRemote/esphomeRemotePlayer.h:
```diff
@@ -73,6 +73,56 @@ std::string filter(std::string str) {
   return output;
 }
 
+std::vector<MenuTitle> parseJsonArray(std::string state, std::string entityId) {
```
I created class TextHelpers and put other related text methods inside it

abroooo reviewed (approved):
I can't try it out as I haven't installed spotcast yet but it looks good 👍
